### DISCUSSION TITLE
How to expose c++ enum class to react native

### DISCUSSION THREAD

**Discussion by ClaudiuHBann** [microsoft/react-native-windows#10624] [Q&A] [answered]
I have a enum class in C++ which looks like this:

```c++
enum class CommandDetectionLevel : uint8_t
  {
    NONE,
    LOW,
    MEDIUM_LOW,
    MEDIUM_HIGH,
    HIGH
  };
```

How can I expose this enum class from a module in react native?

**Comment by jonthysell** [comment, marked as answer]
You would want to expose the enum values as part of a native module's constants. You could specify "CommandDetectionLevel.NONE = 0", "CommandDetectionLevel.LOW = 1", etc.).

See https://microsoft.github.io/react-native-windows/docs/native-modules
